src/arch/xtensa/smp/xtos/xtos-internal.h

```diff
@@ -333,6 +334,29 @@ XTOS_PENDING_OFS:	.space	4	/* _xtos_pending variable */
 	add		\ax, \ax, \ay
 	.endm
 
+	// xtos_stack_addr_percore ax, ay, stack_master, stack_slave, stack_size
+	// Retrieves address of end of stack buffer for certain core to register ax.
+	.macro	xtos_stack_addr_percore ax, ay, stack_master, stack_slave, stack_size
+	get_prid	\ax
+	bnez		\ax, core_s
```

Comment:
Core 0 doesn't need to be master core - that's why we have `PLATFORM_MASTER_CORE_ID`.

Reply:
It is a real scenario or just a theoretical option? I can branch in this macro if core is not a PLATFORM_MASTER_CORE_ID but it will work correctly only if it is 0, just like in memmap in memory.c - it is designed for master at 0

Reply:
Real scenario as there is already HW with master core != 0 (although currently not supported by SOF). I know it probably won't work due to some historical assumptions, but better not to limit the new code.

Reply:
Yes, if we should not assume master == 0 in the generic xtensa arch code since the ID of the core must be unique and that is it, the actual value depends on how the cores are integrated.

We may assume it temporarily (better not, however safe atm) in the cavs platform code since we know how the cores are integrated in there.